### PR TITLE
aws: Add option to provide multiple instance types on cluster creation

### PR DESCRIPTION
**Pull request by hakman**
Alternative to #15849, for use in scalability tests.

/cc @justinsb @johngmyers @dims

### CODE REVIEW

File: upup/pkg/fi/cloudup/new_cluster.go [hunk 1]
```diff
@@ -1055,7 +1066,21 @@ func setupNodes(opt *NewClusterOptions, cluster *api.Cluster, zoneToSubnetsMap m
 			g.Spec.NodeLabels["cloud.google.com/metadata-proxy-ready"] = "true"
 		}
 
-		g.Spec.MachineType = opt.NodeSize
+		for i, size := range opt.NodeSizes {
+			if i == 0 {
```
Comment (dims) [inline upup/pkg/fi/cloudup/new_cluster.go:1070]: same if condition twice?

Comment (hakman) [reply upup/pkg/fi/cloudup/new_cluster.go:1070]: Thanks for catching. Was in a hurry earlier.

**Comment by justinsb** [comment]
As this is for scalability tests, I'm guessing it's not something we would need in 1.27 or 1.28

/milestone v1.29

**Comment by johngmyers** [comment]
@hakman it looks like this broke the kops-aws-cni-cilium-eni periodic. We need to de-duplicate the instance types.

**Comment by hakman** [comment]
@johngmyers https://github.com/kubernetes/test-infra/pull/30664

**Pull request merged**